src/eoapi/raster/eoapi/raster/factory.py

```diff
@@ -46,3 +54,155 @@ def stac_demo(
+                    # Get Total Number of searches rows
+                    query = [
+                        sql.SQL("SELECT count(*) FROM searches"),
+                        *filters,
+                    ]
+                    cursor.execute(sql.SQL(" ").join(query))
+                    nb_items = cursor.fetchone()[0]
+
+                    # Get rows
+                    cursor.row_factory = class_row(model.Search)
+                    query = [
+                        sql.SQL("SELECT * FROM searches"),
+                        *filters,
+                        *offset_and_limit,
+                    ]
+
+                    cursor.execute(sql.SQL(" ").join(query))
+
+                    searches_info = cursor.fetchall()
+
+            qs = QueryParams({**request.query_params, "limit": limit, "offset": offset})
+            links = [
+                model.Link(
+                    rel="self",
+                    href=self.url_for(request, "list_mosaic") + f"?{qs}",
+                ),
+            ]
+
+            if len(searches_info) < int(nb_items):
+                next_token = offset + len(searches_info)
+                qs = QueryParams(
+                    {**request.query_params, "limit": limit, "offset": next_token}
+                )
+                links.append(
+                    model.Link(
+                        rel="next",
+                        href=self.url_for(request, "list_mosaic") + f"?{qs}",
+                    ),
+                )
+
+            if offset > 0:
+                prev_token = offset - limit if (offset - limit) > 0 else 0
+                qs = QueryParams(
+                    {**request.query_params, "limit": limit, "offset": prev_token}
+                )
+                links.append(
+                    model.Link(
+                        rel="prev",
+                        href=self.url_for(request, "list_mosaic") + f"?{qs}",
+                    ),
+                )
+
+            return Infos(
+                searches=[
+                    model.Info(
+                        search=search,
+                        links=[
+                            model.Link(
+                                rel="metadata",
+                                href=self.url_for(
+                                    request, "info_search", searchid=search.id
+                                ),
+                            ),
+                            model.Link(
+                                rel="tilejson",
+                                href=self.url_for(
+                                    request, "tilejson", searchid=search.id
+                                ),
+                            ),
+                        ],
+                    )
+                    for search in searches_info
+                ],
+                links=links,
+                numberMatched=int(nb_items),
+                numberReturned=len(searches_info),
+            )
```

Comment:
@bitner can you check that I'm not doing any stupid SQL (or python, or general) stuff.

Reply:
Nothing bad here that needs to change to merge, but some suggestions on best practices above.